src/oic/utils/authn/client.py

```diff
@@ -121,8 +121,21 @@ def construct(self, cis, request_args=None, http_args=None, **kwargs):
         if "headers" not in http_args:
             http_args["headers"] = {}
 
-        credentials = "{}:{}".format(quote_plus(user), quote_plus(passwd))
-        authz = base64.b64encode(credentials.encode("utf-8")).decode("utf-8")
+        if "encoding" in kwargs:
+            encoding = kwargs["encoding"]
+        else:
+            encoding = "utf-8"
+
+        if "url_encoded" in kwargs:
+            url_encoded = kwargs["url_encoded"]
+        else:
+            url_encoded = True
```

Comment:
I would rather define the kwargs with their default values in the method signature (and also document in the docstring).

Reply:
Hm, it looks like adding the annotation triggers complains from `mypy`. I am not sure if there is an easy way out of here.
Will have a look.

There are some other quality issues that need to be solved.

Reply:
OK, the mypy complain is because of the `cis` arg that is not in the superclass. One way to fix this is to remove the typing annotations. The other way is to make the signature the same (by making the `cis` a kwarg rather than arg).

I would go with the latter. If you are not willing to do that, make a separate issue and I will deal with that and you can base your PR on top of that.